src/collection.ts

```diff
@@ -261,6 +259,13 @@ export class Collection implements OperationParent {
     return this.s.readPreference;
   }
 
+  /**
+   * The current topology of the collection.
+   */
+  get topology(): Topology | undefined {
```

Comment:
This here exposes `topology` as a top-level API, I'm generally OK with this, but it is something to acknowledge. We're usually conscious about what we allow as apart of the stable API. We can't make it private because we need to call it internally `collection.topology`. Calling the `.s.properties` is pretty annoying. Our users have always been able to call `.s.topology` but I don't think we consider that stable. 

> Side thought may not need breaking change `!` here?
 
I really like it, and I'd also be interested to add getters in all the places we're using topology / parent / db / client. 

```ts
  get db () {
    return this.s.db;
  }

 get topology () {
    return this.db.topology;
  }
```

This would help in that we can always just call `this.topology` from within a given class.

Reply:
I think the getters idea is really good! Am happy to add that.

Reply:
Hm, I'm not sure we should take this approach. My main question is: who is this feature helping? Do we expect users to need to refer to the `topology` or the `db` of a `Collection`, or are we adding something here that only helps us (the maintainers)?

For starters, I don't think we should make it easy to access the `Topology` type, since we want that to be private API that we can change at will. Leaving the topology in the `s` is unsavory, but at least let's us make decisions later about how we want to continue to shield users from accessing private implementation.

If it's too hard to stomach `this.s.db.client.topology`, we could consider a new `getTopology` util method. Given how boilerplate this all is, I'm not sure it's worth the effort to make it look clean though.

Reply:
This is a good point. It seems like this would just be for the maintainers' benefit.

One thing to note is that a `topology` getter already exists on the `Db` class. Is this something we would consider removing?

Reply:
For now I have gone with a `getTopology` util method on the `Collection` class. Since a `topology` field exists on `MongoClient` and a `topology` getter exists on `Db`, it's a bit inconsistent. To address this, I think we could remove the `topology` getter on `Db` and (potentially) replace it with a `getTopology` util.